### PR TITLE
Added is_whole boolean for is_number checks

### LINKED ISSUE #220: Add is_whole boolean for is_number checks
Three of the syntax error checks that call out expecting a whole number use the is_number check, which is allowing floats to be used in those fields.  Add a "is_whole" boolean to those checks, which defaults to False in the is_number check, so that only isdigit() is used to determine correctness of the value type.

### PR DESCRIPTION
**Pull request by jcadam14**
Closes #220

### CODE REVIEW

File: src/regtech_data_validator/check_functions.py [hunk 1]
```diff
@@ -318,7 +318,10 @@ def is_number(ct_value: str, accept_blank: bool = False) -> bool:
     Returns:
         bool: True if value is number , False if value is not number
     """
-    value_check = ct_value.isdigit() or bool(re.match(r"^[-+]?[0-9]*\.?[0-9]+$", ct_value))
+    if is_whole:
+        value_check = ct_value.isdigit()
+    else:
+        value_check = ct_value.isdigit() or bool(re.match(r"^[-+]?[0-9]*\.?[0-9]+$", ct_value))
```
Comment (hkeeler) [inline src/regtech_data_validator/check_functions.py:324]: I know the regex bits were already there, but something like this seems like simpler logic.

```suggestion
    value_check = True
    num_parser = int if is_whole else float
    try:
        num_parser(ct_value)
    except ValueError:
        value_check = False
```

Also, `str.isdigit` probably isn't the best check. You can put funky stuff in there like:

```python
>>> '2²'.isdigit()
True
```

I think `str.isdecimal` is the more appropriate function...if we want to stick with that pattern:

```python
>>> '2²'.isdecimal()
False
>>> '2.0'.isdecimal()
False
>>> '2'.isdecimal()
True
```

File: src/regtech_data_validator/check_functions.py [hunk 2]
```diff
@@ -318,7 +318,18 @@ def is_number(ct_value: str, accept_blank: bool = False) -> bool:
     Returns:
         bool: True if value is number , False if value is not number
     """
-    value_check = ct_value.isdigit() or bool(re.match(r"^[-+]?[0-9]*\.?[0-9]+$", ct_value))
+
+    def num_check():
```
Comment (lchen-2101) [inline src/regtech_data_validator/check_functions.py]: if we want to abstract the logic into its own function, I'd rather see it outside of the `is_number` function

**Review by hkeeler — APPROVED** [review]
LGTM. All the tests still pass. 😄 My only concern was just that I may have missed some edge case the previous logic was handling.

**Comment by jcadam14** [comment]
> LGTM. All the tests still pass. 😄 My only concern was just that I may have missed some edge case the previous logic was handling.

Not using the pytests, I tested with several sblars I've generated with various values for fields using this check and they all worked as expected.  So functional tests gave me the piece of mind I needed for this ;)

**Comment by hkeeler** [comment]
I knew you would. 😉 ...and thank you for doing so. 😄 I also tested a fair bit locally, but just at the function level. Nothing like using real files.
